hyperspy/io_plugins/tiff.py

```diff
@@ -275,185 +289,483 @@ def _load_data(serie, is_rgb, sl=None, memmap=None, **kwds):
+            scales['z'] = imagej_metadata['spacing']
+    return scales, offsets, units
+
+
+def _is_digital_micrograph(op) -> bool:
+    # for files containing DM metadata
+    tags = ['65003', '65004', '65005', '65009', '65010', '65011',
+            '65006', '65007', '65008', '65022', '65024', '65025']
+    search_result = [tag in op for tag in tags]
+    return any(search_result)
+
+
+def _intensity_axis_digital_micrograph(op, intensity_axis = {}):
+    if '65022' in op:
+        intensity_axis['units'] = op['65022']  # intensity units
+    if '65024' in op:
+        intensity_axis['offset'] = op['65024']  # intensity offset
+    if '65025' in op:
+        intensity_axis['scale'] = op['65025']  # intensity scale
+    return intensity_axis
+
+
+def _add_axes_digital_micrograph(op, scales, offsets, units ):
+    if '65003' in op:
+        _logger.debug("Reading Gatan DigitalMicrograph tif metadata")
+        units['y'] = op['65003']  # x units
+    if '65004' in op:
+        units['x'] = op['65004']  # y units
+    if '65005' in op:
+        units['z'] = op['65005']  # z units
+    if '65009' in op:
+        scales['y'] = op['65009']  # x scales
+    if '65010' in op:
+        scales['x'] = op['65010']  # y scales
+    if '65011' in op:
+        scales['z'] = op['65011']  # z scales
+    if '65006' in op:
+        offsets['y'] = op['65006']  # x offset
+    if '65007' in op:
+        offsets['x'] = op['65007']  # y offset
+    if '65008' in op:
+        offsets['z'] = op['65008']  # z offset
+
+    return scales, offsets, units
+
+
+def _parse_scale_unit(tiff, page, op, shape, force_read_resolution, names, **kwds):
+    # Force reading always has priority
+    if _is_force_readable(op, force_read_resolution):
+        axes = _axes_force_read(op, shape, names)
+        return axes
+    # Other axes readers can change position if you need to do it
+    elif _is_fei(tiff):
+        axes = _axes_fei(tiff, op, shape, names)
+        return axes
+    elif _is_zeiss(tiff):
+        axes = _axes_zeiss(tiff, op, shape, names)
+        return axes
+    elif _is_tvips(tiff):
+        axes = _axes_tvips(tiff, op, shape, names)
+        return axes
+    elif _is_olympus_sis(page):
+        axes = _axes_olympus_sis(page, tiff, op, shape, names)
+        return axes
+    elif _is_jeol_sightx(op):
+        axes = _axes_jeol_sightx(tiff, op, shape, names)
+        return axes
+    elif _is_streak_hamamatsu(op):
+        axes = _axes_hamamatsu_streak(tiff, op, shape, names, **kwds)
+        return axes
+    # Axes are otherwise set to defaults
+    else:
+        scales, offsets, units = _axes_defaults()
+        # Axes descriptors can be additionally parsed from digital micrograph or imagej-style files
+        if _is_digital_micrograph(op):
+            scales, offsets, units = _add_axes_digital_micrograph(op, scales, offsets, units)
+        if _is_imagej(tiff):
+            scales, offsets, units = _add_axes_imagej(tiff, op, scales, offsets, units)
```

Comment:
It seems to me that the tiff loader/writer plugin was written to support digital micrograph or imagej metadata on top of other file formats. 

I think the usecase would be loading an FEI file with FEI axes calibrations, but with some calibration aspects, e.g. one axis or a noise model (see _test_read_unit_from_dm in the test_tiff.py for eyample), calibrated using dm3 or imagej-style metadata.

The problem is that not all the metadata parsers are supporting this, so I found it difficult to address this in a single level of code...

Reply:
> It seems to me that the tiff loader/writer plugin was written to support digital micrograph or imagej metadata on top of other file formats.
>
> I think the usecase would be loading an FEI file with FEI axes calibrations, but with some calibration aspects, e.g. one axis or a noise model (see _test_read_unit_from_dm in the test_tiff.py for eyample), calibrated using dm3 or imagej-style metadata.

I am not sure, it may simply have been changed over time in an inconsistent manner...
 
> The problem is that not all the metadata parsers are supporting this, so I found it difficult to address this in a single level of code...

This is why, I would remove it.